pkg/pillar/cmd/zedagent/handlemetrics.go

```diff
@@ -1544,6 +1544,72 @@ func PublishContentInfoToZedCloud(ctx *zedagentContext, uuid string,
 	}
 }
 
+// PublishVolumeToZedCloud is called per change, hence needs to try over all management ports
+// When volume status is nil it means a delete and we send a message
+// containing only the UUID to inform zedcloud about the delete.
+func PublishVolumeToZedCloud(ctx *zedagentContext, uuid string,
+	volStatus *types.VolumeStatus, iteration int) {
+
+	log.Infof("PublishVolumeToZedCloud uuid %s", uuid)
+	var ReportInfo = &info.ZInfoMsg{}
+
+	volumeType := new(info.ZInfoTypes)
+	*volumeType = info.ZInfoTypes_ZiVolume
+	ReportInfo.Ztype = *volumeType
+	ReportInfo.DevId = *proto.String(zcdevUUID.String())
+	ReportInfo.AtTimeStamp = ptypes.TimestampNow()
+
+	ReportVolumeInfo := new(info.ZInfoVolume)
+
+	ReportVolumeInfo.Uuid = uuid
+	ReportVolumeInfo.State = info.ZSwState_HALTED
+	if volStatus != nil {
+		ReportVolumeInfo.DisplayName = volStatus.DisplayName
+		ReportVolumeInfo.State = volStatus.State.ZSwState()
+
+		if !volStatus.ErrorTime.IsZero() {
+			errInfo := encodeErrorInfo(
+				volStatus.ErrorAndTimeWithSource.ErrorAndTime())
+			ReportVolumeInfo.VolumeErr = errInfo
+		}
+
+		ReportVolumeInfo.ProgressPercentage = uint32(volStatus.Progress)
```

Comment:
Don't we also have some size to report e.g., from qemu-img setting the actual size and the max size?

Reply:
Yes, I will add them.

Reply:
Resolved.